polygeist/tools/cgeist/Lib/CGCall.cc

```diff
@@ -1522,8 +1522,16 @@ MLIRScanner::emitBuiltinOps(clang::CallExpr *Expr) {
     V = Builder.create<math::PowFOp>(Loc, Args[0], Args[1]);
   } break;
   case clang::Builtin::BI__builtin_assume: {
-    VisitArgs();
-    Builder.create<LLVM::AssumeOp>(Loc, Args[0]);
+    if (!Expr->getArg(0)->HasSideEffects(Glob.getTypes().getContext())) {
+      VisitArgs();
+      Builder.create<LLVM::AssumeOp>(Loc, Args[0]);
+    }
+    // Early return as we have two possible scenarios here:
+    // 1. We have not generated the intrinsic, but we don't want the call to
+    // __builtin_assume to be generated either.
+    // 2. We have generated the intrinsic and we don't want the call to
+    // __builtin_assume to be generated.
+    return {{}, true};
```

Comment:
We need to handle this differently as the operation generated defines no value.

Reply:
This switch case here handles the `builtin_assume`. I understand that if the argument has a side effects we do not want to generate `builtin_assume`, but what is the "intrinsic" mentioned in the comment ?

Reply:
The `LLVM::AssumeOp` we're creating above these lines

Reply:
ok I c